pkg/telemetry/context.go

```diff
@@ -0,0 +1,53 @@
+package telemetry
+
+import (
+	"context"
+	"strings"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc"
+)
+
+type rpcContextName string
+
+const (
+	methodContextName  rpcContextName = "method"
+	serviceContextName rpcContextName = "service"
+)
+
+func saveMethodServiceInContext(ctx context.Context, method string, service string) context.Context {
+	methods := strings.Split(method, "/")
+
+	ctx = context.WithValue(ctx, methodContextName, methods[len(methods)-1])
+	return context.WithValue(ctx, serviceContextName, service)
+}
+
+// ContextWithRPCInfo will save the rpc method and service information in context
+func ContextWithRPCInfo(ctx context.Context) context.Context {
+	method, found := grpc.Method(ctx)
```

Comment:
This is dependent upon a middleware upstream. The purpose of these context setters and getters is to eliminiate the codependency on the `grpc` transport.

This code should simply change to
```go
func ContextWithRPCInfo(ctx context.Context, info RPCInfo) {
    return context.WithValue(ctx, rpcInfoContextKey, info)
}
```

then there is no co-dependency on the grpc middleware at the transport layer above this point in the code.